test/integration/preflight/preflight_test.go

```diff
@@ -0,0 +1,120 @@
+package preflight
```

Comment:
Can't say I'm a huge fan of using golang for testing here, seems like it is going to be difficult to get external contributors to participate if the tests are written in golang.

I'd much prefer that we move towards a framework like https://molecule.readthedocs.io instead, since it also allows for much more comprehensive testing against multiple machines, etc.